spanner.go

```diff
@@ -293,6 +294,23 @@ func EncodeSpannerStructFields(
 	return nil
 }
 
+func SpannerReadStructAndDecode[I any, T any](ctx context.Context, cli *spanner.Client, table string, key spanner.Key, target *T) error {
+	var dest I
+	row, err := cli.Single().ReadRow(ctx, table, key, SpannerStructFieldNames(dest))
+	if err != nil {
+		if spanner.ErrCode(err) == codes.NotFound {
+			return Errorf(fmt.Errorf("%w: %v", ErrObjectNotFound, err))
```

Comment:
I believe `%w` must be last in fmt string for it to work, but you could add a test to check !